packages/next/build/webpack/loaders/next-flight-server-loader.ts

```diff
@@ -58,29 +67,49 @@ async function parseImportsInfo(
       case 'ImportDeclaration': {
         const importSource = node.source.value
         if (!isClientCompilation) {
+          // Server compilation for .server.js.
+          if (isServerComponent(importSource)) {
+            continue
+          }
+
+          const importDeclarations = source.substring(
+            lastIndex,
+            node.source.span.start - beginPos
+          )
+
           if (
             !(
-              isClientComponent(importSource, pageExtensions) ||
+              isClientComponent(importSource) ||
```

Comment:
Are we able to reduce those conditions? This condition looks like can be replaced with `!isServerComponent(...)`

Reply:
`!isServerComponent(...)` matches client components here too, but we **only** need shared components in this condition. I don't think it's possible to avoid `isClientComponent` being used, can you share a full code change suggestion so I can take a look?

Reply:
I see. My intention was to see if we can remove the `isNextComponent` or `isImageImport` here since they could be handled in a more general logic.

Could we change the `ImportDeclaration` switch handler to

```
if (isServerComponent(..)) return
if (isClientComponent(...)) {
   // handle client component
} else {
   // treat as shared component
}
```

Reply:
Got it, yep. I'm just keeping minimal structure changes in this PR and make it work, can definitely refactor it a bit more.